docs/source/index.rst

```diff
@@ -89,6 +91,33 @@ For the development version, you can do almost the same:
     export GSTOOLS_BUILD_PARALLEL=1
     pip install git+git://github.com/GeoStat-Framework/GSTools.git@main
 
+**Using experimental GSTools-Core for even more speed**
+
+You can install the optional dependency `GSTools-Core <https://github.com/GeoStat-Framework/GSTools-Core>`_,
+which is a re-implementation of the main algorithms used in GSTools. The new
+package uses the language Rust and it should be faster (in some cases by orders
+of magnitude), safer, and it will potentially completely replace the current
+standard implementation in Cython. Once the package GSTools-Core is available
+on your machine, it will be used by default. In case you want to switch back to
+the Cython implementation, you can set :code:`gstools.config.USE_RUST=False` in
+your code.  This also works at runtime. You can install the optional dependency
+e.g. by
+
+.. code-block:: none
+
+    pip install gstools[rust]
+
+or by manually installing the package
+
+.. code-block:: none
+
+    pip install gstools-core
+
+GSTools-Core will automatically use all your cores in parallel, without having
+to use OpenMP or a local C compiler.
+In case you want to restrict the number of threads used, you can set the
+environment variable ``RAYON_NUM_THREADS`` to the desired amount.
```

Comment:
What about a little function in `gstools.config` to set this environment variable?

Reply:
One argument against that is that this variable will affect all Rust software using Rayon, not just GSTools-Core's backend.

If we do provide a programmatic interface, we should probably make the Core initialize a dedicated thread pool and make this function affect only this thread pool?

Reply:
@adamreichold raises a fair point there. But that env. variable would only exist during the shell session from which GSTools is run (I don't have enough experience with Windows to know how it handles env. variables).
Did I understand it correctly, that the number of threads used by Rayon can only be set once at the start and doing it a second time results in an error? I'm not if this is worth the effort.
Another alternative would be to capture the env. variable at the start of GSTools and set it back to that value when exiting, but that doesn't sound very elegant.
I personally would just keep the hint in the readme for the moment. But if you have stronger opinions than I do, I'd be happy to implement your preferred solution.

Reply:
> Did I understand it correctly, that the number of threads used by Rayon can only be set once at the start and doing it a second time results in an error? I'm not if this is worth the effort.

Setting the environment variable will affect Rayon's so-called [global](https://docs.rs/rayon/1.5.1/rayon/struct.ThreadPoolBuilder.html#method.build_global) thread pool which is initialized once, either on-demand when the first Rayon task is spawn or explicitly be the application. Hence, changing the environment variable after a first computation using Rayon was run has no effect.

This why I suggested making the GSTools-Core module store an `Option<ThreadPool>` which would be affected (meaning re-initialized using a given number of threads) by the function from `gstools.config`. If the option is set, that thread pool is used via [`ThreadPool::install`](https://docs.rs/rayon/1.5.1/rayon/struct.ThreadPool.html#method.install), otherwise GSTools-Core falls back to the global thread pool.

> I personally would just keep the hint in the readme for the moment.

Personally, I would say that online control over parallelism is a somewhat orthogonal issue and would best be served via a follow-up PR / issue.